src/browser/simulatedstore/alertHelper.ts

```diff
@@ -0,0 +1,81 @@
+import { Alert } from "react-native";
+import { PurchasesPackage } from "@revenuecat/purchases-typescript-internal";
+import { loadSimulatedPurchaseData, SimulatedPurchaseData } from "./offeringsLoader";
+
+/**
+ * Shows a simulated purchase alert for platforms that don't support DOM manipulation.
+ */
+export function showSimulatedPurchaseAlert(
+  packageIdentifier: string, 
+  offeringIdentifier: string,
+  onPurchase: (packageInfo: PurchasesPackage) => Promise<void>,
+  onCancel: () => void
+): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const handleCancel = () => {
+      onCancel();
+      resolve();
+    };
+
+    const handlePurchase = async (data: SimulatedPurchaseData) => {
+      try {
+        await onPurchase(data.packageInfo);
+        resolve();
+      } catch (error) {
+        reject(error);
+      }
+    };
+
+    loadSimulatedPurchaseData(packageIdentifier, offeringIdentifier)
+      .then((data) => {
+        const { packageInfo, offers } = data;
+
+        let message = `⚠️ This is a test purchase and should only be used during development. In production, use an Apple/Google API key from RevenueCat.\n\n`;
+        message += `Package ID: ${packageInfo.identifier}\n`;
+        message += `Product ID: ${packageInfo.product.identifier}\n`;
+        message += `Title: ${packageInfo.product.title}\n`;
+        message += `Price: ${packageInfo.product.priceString}\n`;
+
+        if (packageInfo.product.subscriptionPeriod) {
+          message += `Period: ${packageInfo.product.subscriptionPeriod}\n`;
+        }
+
+        if (offers.length > 0) {
+          message += `\nOffers:\n${offers.map(offer => `• ${offer}`).join('\n')}`;
+        }
+
+        Alert.alert(
+          'Test Purchase',
+          message,
+          [
+            {
+              text: 'Cancel',
+              style: 'cancel',
+              onPress: handleCancel,
+            },
+            {
+              text: 'Test Purchase',
+              onPress: () => handlePurchase(data),
+            },
+          ],
+          { cancelable: true, onDismiss: handleCancel }
+        );
+      })
+      .catch((error) => {
+        console.error('Error loading package details:', error);
+        const errorMessage = error instanceof Error ? error.message : 'Unknown error';
+
+        Alert.alert(
+          'Test Purchase Error',
+          `Error loading package details: ${errorMessage}`,
+          [
+            {
+              text: 'Close',
+              onPress: handleCancel,
```

Comment:
If I'm following the code path correctly, I think this would show an alert saying "User cancelled purchase.", which would be weird in the case of an error

Reply:
Hmm well, this would make the `purchasePackage` promise fail with an error code `1` and a message indicating that the user cancelled the purchase, which I think is what we do on mobile for cancellations if I'm correct? There is no alert with a "User cancelled purchase" I believe

Reply:
Talked over Slack and we were indeed failing with the wrong error (user cancelled even though it wasn't a user cancellation). Fix here: https://github.com/RevenueCat/react-native-purchases/pull/1353/commits/5f42105e82d4948393ed437844f297b5bc22ff83